sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/models/CallMediaRecognizeOptions.java

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callautomation.models;
+
+import com.azure.communication.common.CommunicationIdentifier;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.time.Duration;
+
+/** Options to configure the Recognize operation **/
+public abstract class CallMediaRecognizeOptions {
+    /*
+     * Determines the type of the recognition.
+     */
+    @JsonProperty(value = "recognizeInputType", required = true)
+    private RecognizeInputType recognizeInputType;
+
+    /*
+     * The source of the audio to be played for recognition.
+     */
+    @JsonProperty(value = "playPrompt")
+    private PlaySource playPrompt;
+
+    /*
+     * If set recognize can barge into other existing
+     * queued-up/currently-processing requests.
+     */
+    @JsonProperty(value = "interruptCallMediaOperation")
+    private Boolean interruptCallMediaOperation;
```

Comment:
what is the reason behind `Boolean` instead of `bool`?

Reply:
That seems to be the convention in the sdk. Do you think I should use boolean instead?

Reply:
nah, if other SDKs have it, leave it like that. Let's test it what it assigns to it when sending the request.